app/models/user.rb

```diff
@@ -37,6 +37,15 @@ class User < ApplicationRecord
   validates_acceptance_of :terms_and_conditions
   validates_confirmation_of :password, on: :create
   validates :currency, inclusion: { in: ExchangeRateService::CURRENCIES }
+  validate :email_uniqueness
+
+  protected
+
+  def email_uniqueness
+    if User.all.map(&:email).include?(email) && !persisted?
+      errors.add(:email, :taken)
+    end
+  end
```

Comment:
We actually don't need email uniqueness, since only username is used for login and unique identification.

Validating this will only create a public endpoint to check for existing users' email presence.  I could try to signup using joe@shmo.com and see if it returns this validation error.  If it does, I know that email is being used in the system.

Please remove this. :)